mod-source-record-storage-server/src/main/java/org/folio/rest/impl/SourceStorageMigrationsJobsImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.folio.dataimport.util.ExceptionHelper;
+import org.folio.rest.jaxrs.model.AsyncMigrationJobInitRq;
+import org.folio.rest.jaxrs.resource.SourceStorageMigrationsJobs;
+import org.folio.services.migrations.AsyncMigrationJobService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public class SourceStorageMigrationsJobsImpl implements SourceStorageMigrationsJobs {
+
+  private static final String NOT_FOUND_MSG = "Async migration job with id '%s' was not found";
+
+  private final String tenantId;
+
+  @Autowired
+  private AsyncMigrationJobService asyncMigrationJobService;
+
+  public SourceStorageMigrationsJobsImpl(Vertx vertx, String tenantId) { //NOSONAR
+    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
+    this.tenantId = tenantId;
+  }
+
+  @Override
+  public void postSourceStorageMigrationsJobs(AsyncMigrationJobInitRq entity, Map<String, String> okapiHeaders,
```

Comment:
Does this mean any user can initiate a migration?

Reply:
only users which have "source-storage.migrations.post" permission can initiate migration